client/state/sites/domains/assembler.js

```diff
@@ -79,6 +79,10 @@ export const createSiteDomainObject = ( domain ) => {
 		isRedeemable: Boolean( domain.is_redeemable ),
 		isEligibleForInboundTransfer: Boolean( domain.is_eligible_for_inbound_transfer ),
 		isAutoRenewing: Boolean( domain.auto_renewing ),
+		isIcannVerificationSuspended:
+			typeof domain.is_icann_verification_suspended === 'boolean'
+				? Boolean( domain.is_icann_verification_suspended )
+				: null,
```

Comment:
Suggestion: a shorter but kinda hacky way of doing this is `!! domain.is_icann_verificatino_suspended` - what do you think? 🙂 It's not exactly the same because a non-boolean value would get transformed into one, but I guess it wouldn't be a problem in the contexts where this is used.